src/ec/curve25519/x25519.rs

```diff
@@ -78,9 +78,18 @@ fn x25519_ecdh(out: &mut [u8], my_private_key: &ec::PrivateKey,
     let peer_public_key =
         try!(slice_as_array_ref!(peer_public_key.as_slice_less_safe(),
                                  PUBLIC_KEY_LEN));
-    bssl::map_result(unsafe {
-        GFp_x25519_ecdh(out, my_private_key, peer_public_key)
-    })
+
+    unsafe {
+        GFp_x25519_scalar_mult(out, my_private_key, peer_public_key);
+    }
+
+    let zeros: SharedSecret = [0; SHARED_SECRET_LEN];
+    if constant_time::verify_slices_are_equal(out, &zeros).is_ok() {
```

Comment:
Yeah, looking forward to doing something better here. I started out adding a new `constant_time` function to test for zero, but thought it best to start with the simplest thing that was obviously correct.